File: testBack/src/main/java/com/bookadmin/modules/category/entity/Category.java
package com.bookadmin.modules.category.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 图书分类,与前端 src/types/category.ts 对应。
 */
@Data
@TableName("category")
public class Category {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private Long parentId;
    private Integer sort;
    private String description;
    private Integer bookCount;
    private LocalDateTime createdAt;
}

File: test/src/types/category.ts
// 图书分类类型

export interface Category {
  id: number;
  name: string;
  parentId: number | null;
  sort: number;
  description: string;
  bookCount: number;
  createdAt: string;
}

/** 分类新增/编辑表单 */
export type CategoryForm = Omit<Category, 'id' | 'bookCount' | 'createdAt'>;
